--- origsrc/binutils-2.35.1/bfd/configure.ac
+++ src/binutils-2.35.1/bfd/configure.ac
@@ -320,8 +320,8 @@ changequote([,])dnl
   case "${host}" in
   # More hacks to build DLLs on Windows.
   *-*-cygwin*)
-    SHARED_LDFLAGS="-no-undefined"
-    SHARED_LIBADD="-L`pwd`/../libiberty -liberty -L`pwd`/../intl -lintl -lcygwin -lkernel32"
+    SHARED_LDFLAGS="-no-undefined -Wl,--exclude-libs=libiberty.a"
+    SHARED_LIBADD="-Wl,`pwd`/../libiberty/libiberty.a -L`pwd`/../intl -lintl -lcygwin -lkernel32"
   ;;
 
   # Use built-in libintl on macOS, since it is not provided by libc.
--- origsrc/binutils-2.35.1/opcodes/configure.ac
+++ src/binutils-2.35.1/opcodes/configure.ac
@@ -175,8 +175,8 @@ changequote([,])dnl
 
   case "${host}" in
     *-*-cygwin*)
-      SHARED_LDFLAGS="-no-undefined"
-      SHARED_LIBADD="-L`pwd`/../bfd -lbfd -L`pwd`/../libiberty -liberty -L`pwd`/../intl -lintl -lcygwin"
+      SHARED_LDFLAGS="-no-undefined -Wl,--exclude-libs=libiberty.a"
+      SHARED_LIBADD="`pwd`/../bfd/libbfd.la -Wl,`pwd`/../libiberty/libiberty.a -L`pwd`/../intl -lintl -lcygwin"
       ;;
    *-*-darwin*)
      SHARED_LIBADD="-Wl,`pwd`/../bfd/.libs/libbfd.dylib ${SHARED_LIBADD}"
--- origsrc/binutils-2.35.1/bfd/configure
+++ src/binutils-2.35.1/bfd/configure
@@ -14584,8 +14584,9 @@ if test "$enable_shared" = "yes"; then
   case "${host}" in
   # More hacks to build DLLs on Windows.
   *-*-cygwin*)
-    SHARED_LDFLAGS="-no-undefined"
-    SHARED_LIBADD="-L`pwd`/../libiberty -liberty -L`pwd`/../intl -lintl -lcygwin -lkernel32"
+    SHARED_LDFLAGS="-no-undefined -Wl,--exclude-libs=libiberty.a"
+    SHARED_LIBADD="-Wl,`pwd`/../libiberty/libiberty.a -L`pwd`/../intl -lintl -lcygwin -lkernel32"
+
   ;;
 
   # Use built-in libintl on macOS, since it is not provided by libc.
--- origsrc/binutils-2.35.1/opcodes/configure
+++ src/binutils-2.35.1/opcodes/configure
@@ -12784,8 +12784,9 @@ if test "$enable_shared" = "yes"; then
 
   case "${host}" in
     *-*-cygwin*)
-      SHARED_LDFLAGS="-no-undefined"
-      SHARED_LIBADD="-L`pwd`/../bfd -lbfd -L`pwd`/../libiberty -liberty -L`pwd`/../intl -lintl -lcygwin"
+      SHARED_LDFLAGS="-no-undefined -Wl,--exclude-libs=libiberty.a"
+      SHARED_LIBADD="`pwd`/../bfd/libbfd.la -Wl,`pwd`/../libiberty/libiberty.a -L`pwd`/../intl -lintl -lcygwin"
+
       ;;
    *-*-darwin*)
      SHARED_LIBADD="-Wl,`pwd`/../bfd/.libs/libbfd.dylib ${SHARED_LIBADD}"
